src/tree/FunctionAppProvider.ts

```diff
@@ -78,8 +78,24 @@ export class FunctionAppProvider implements IChildProvider {
         const actionContext: IActionContext = userOptions ? userOptions.actionContext : <IActionContext>{ properties: {}, measurements: {} };
         const resourceGroup: string | undefined = userOptions ? userOptions.resourceGroup : undefined;
         const runtime: ProjectRuntime = await getDefaultRuntime(actionContext);
-        const appSettings: { [key: string]: string } = await getCliFeedAppSettings(runtime);
-        const site: Site = await createFunctionApp(actionContext, parent, showCreatingNode, appSettings, resourceGroup);
+        const functionAppSettings: { [key: string]: string } = await getCliFeedAppSettings(runtime);
+        const language: string | undefined = getFuncExtensionSetting(projectLanguageSetting);
+        const createOptions: IAppCreateOptions = { functionAppSettings, resourceGroup };
+
+        if (language === ProjectLanguage.Python) {
+            createOptions.os = 'linux';
+            createOptions.runtime = 'python';
+        } else {
+            // If the language isn't set, just assume it's not Python. Python is still in preview and it's not worth an extra prompt yet
+
+            // Default to windows because linux is still in preview
```

Comment:
the os can't be overridden. I guess it might be more accurate to say "_Use_ windows because linux is still in preview"

`WEBSITE_RUN_FROM_PACKAGE` can easily be overridden by the user. It's just an app setting.